mail/libraries/lite_to_edifact_converter.py

```
@@ -155,13 +157,20 @@ def licences_to_edifact(licences: QuerySet, run_number: int) -> str:
     if errors:
         logging.error(f"File content not as per specification, {errors}")
         logging.info(f"Generated file content: {edifact_file}")
-        raise EdifactValidationError
+        raise EdifactValidationError(f"Errors found in the Edifact file - {edifact_file}", errors=errors)
 
     logging.debug(f"Generated file content: {edifact_file}")
 
     return edifact_file
 
 
+def process_edifact_line(line, line_no):
+    """Tries to validate a line before adding it to the EDIFACT file.
+    If the validation fails, skips the line so that we can process the
+    rest of the licenses.
+    """
+    pass
```

I started writing this but there is more to it...

1) We should probably have a function called `get_license_edifact`. This should create all the EDIFACT lines for a particular license and called inside the loop. 

2) `get_license_edifact` should validate the EDIFACT blob produced for each license. If it cannot produce a valid blob, it should raise an exception. On the other hand, if it produces a valid blob, the should mark the license as `processed` **inside the loop**. instead of the following (up the stack) -

https://github.com/uktrade/lite-hmrc/blob/dcccd1cec37acbeb9910ca47e4f529baed41f2a4/mail/libraries/builders.py#L134-L143